File: ios/RNContentBlockCarouselViewManager.m
//
//  RNContentBlockCarouselViewManager.m
//  Exponea
//
//  Created by Adam Mihalik on 10/02/2025.
//  Copyright © 2025 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNContentBlockCarouselViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(initProps, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(overrideDefaultBehavior, BOOL)
RCT_EXPORT_VIEW_PROPERTY(trackActions, BOOL)
RCT_EXPORT_VIEW_PROPERTY(customFilterActive, BOOL)
RCT_EXPORT_VIEW_PROPERTY(customSortActive, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onDimensChanged, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onContentBlockEvent, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onContentBlockDataRequestEvent, RCTDirectEventBlock)
RCT_EXTERN_METHOD(filterResponse:(nonnull NSNumber*)reactTag args:(NSArray *)args)
RCT_EXTERN_METHOD(sortResponse:(nonnull NSNumber*)reactTag args:(NSArray *)args)
@end

File: ios/RNContentBlockCarouselViewManager.swift
//
//  RNContentBlockCarouselViewManager.swift
//  Exponea
//
//  Created by Adam Mihalik on 10/02/2025.
//  Copyright © 2025 Facebook. All rights reserved.
//

import Foundation
import React
import UIKit
import ExponeaSDK

@objc(RNContentBlockCarouselViewManager)
class RNContentBlockCarouselViewManager: RCTViewManager {
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    override func view() -> UIView! {
        return CarouselInAppContentBlockViewProxy()
    }
    @objc(filterResponse:args:)
    func filterResponse(tag: NSNumber, args: NSArray) {
        findNativeView(ofType: CarouselInAppContentBlockViewProxy.self, byTag: tag) { nativeView in
            nativeView.onFilterResponse(args)
        }
    }
    @objc(sortResponse:args:)
    func sortResponse(tag: NSNumber, args: NSArray) {
        findNativeView(ofType: CarouselInAppContentBlockViewProxy.self, byTag: tag) { nativeView in
            nativeView.onSortResponse(args)
        }
    }
}
